### PR TITLE
[Issue #37]: FileItem background colour

### PR DESCRIPTION
**Pull request by EfthymisB**
Moved FiltItem's background colour to `ui_config_advanced`.

### CODE REVIEW

**Comment by EfthymisB** [comment]
It also resolves https://github.com/shotgunsoftware/tk-multi-breakdown2/issues/45.

**Review by rob-aitchison — CHANGES_REQUESTED** [review]
Hello @EfthymisB ,
I am not sure the style matches by introducing the _APP and then making other variables relative and all UPPERCASE. Is this a convention specific to your studio? Please let us know, thanks!

File: hooks/ui_config_advanced.py [hunk 1]
```diff
@@ -313,6 +313,11 @@ def get_item_subtitle(self, index):
 
         return subtitle
 
+    @staticmethod
+    def get_item_background_color():
+        """Returns the brush to use draw the background for this widget"""
+        return QtGui.QApplication.palette().light()
```
Comment (staceyoue) [inline hooks/ui_config_advanced.py:319]: The default background color should not change from what it currently is, so it does not affect other apps using it. Please revert to returning the value:
```
QtGui.QApplication.palette().midlight()
```

If you want to use a different color, you need to override this hook function.

File: python/tk_multi_breakdown2/file_model.py [hunk 2]
```diff
@@ -291,7 +296,7 @@ def data(self, role):
             """
 
             if role == QtCore.Qt.BackgroundRole:
-                return QtGui.QApplication.palette().midlight()
+                return UI_CONFIG_ADV_HOOK.get_item_background_color()
```
Comment (staceyoue) [inline python/tk_multi_breakdown2/file_model.py:299]: Since this is only returning the hook function's return value, we can simplify this by removing this `if` case and add the following line to the `role_methods` dictionary (defined on line `517`):
```
self.role_methods = {
    ...
    QtCore.Qt.BackgroundRole: UI_CONFIG_ADV_HOOK.get_item_background_color
}
```

File: hooks/ui_config_advanced.py (hunk 1 above)
Comment (EfthymisB) [reply hooks/ui_config_advanced.py]: True true, I've updated!

File: python/tk_multi_breakdown2/file_model.py (hunk 2 above)
Comment (EfthymisB) [reply python/tk_multi_breakdown2/file_model.py]: Yeah that's better indeed! I've updated!

File: hooks/ui_config_advanced.py [hunk 3]
```diff
@@ -313,6 +313,11 @@ def get_item_subtitle(self, index):
 
         return subtitle
 
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (EfthymisB) [reply hooks/ui_config_advanced.py]: Hi @rob-aitchison,

I've reply on the other comment so I'll use this space. Yes this is convention we use in our studio but feel free to suggest a different style so it matches your style. I'll be happy to update the code.

File: python/tk_multi_breakdown2/file_model.py [hunk 4]
```diff
@@ -32,6 +32,13 @@
 ViewItemRolesMixin = delegates.ViewItemRolesMixin
 
 
+_APP = sgtk.platform.current_bundle()
+
+# Get the hook instance for configuring the display for model view items.
+ui_config_adv_hook_path = _APP.get_setting("hook_ui_config_advanced")
+UI_CONFIG_ADV_HOOK = _APP.create_hook_instance(ui_config_adv_hook_path)
+
+
```
Comment (staceyoue) [inline python/tk_multi_breakdown2/file_model.py:41]: Please revert this change - it is just a different style convention

File: python/tk_multi_breakdown2/file_model.py [hunk 5]
```diff
@@ -466,7 +468,7 @@ def __init__(self, parent, bg_task_manager, group_by=None, polling=False):
 
         QtGui.QStandardItemModel.__init__(self, parent)
 
-        self._app = sgtk.platform.current_bundle()
+        self._app = _APP
```
Comment (staceyoue) [inline python/tk_multi_breakdown2/file_model.py:471]: Please revert this change

File: python/tk_multi_breakdown2/file_model.py [hunk 6]
```diff
@@ -40,8 +47,6 @@ class FileModel(QtGui.QStandardItemModel, ViewItemRolesMixin):
     File items are grouped into groups defined by the app configuration.
     """
 
-    UI_CONFIG_ADV_HOOK_PATH = "hook_ui_config_advanced"
```
Comment (staceyoue) [inline python/tk_multi_breakdown2/file_model.py:43]: Please revert this change

File: python/tk_multi_breakdown2/file_model.py [hunk 7]
```diff
@@ -507,21 +509,18 @@ def __init__(self, parent, bg_task_manager, group_by=None, polling=False):
         # Add additional roles defined by the ViewItemRolesMixin class.
         self.NEXT_AVAILABLE_ROLE = self.initialize_roles(self.NEXT_AVAILABLE_ROLE)
 
-        # Get the hook instance for configuring the display for model view items.
-        ui_config_adv_hook_path = self._app.get_setting(self.UI_CONFIG_ADV_HOOK_PATH)
-        ui_config_adv_hook = self._app.create_hook_instance(ui_config_adv_hook_path)
```
Comment (staceyoue) [inline python/tk_multi_breakdown2/file_model.py:512]: Please revert this change

**Review by staceyoue — CHANGES_REQUESTED** [review]
@EfthymisB thanks for this contribution 👍 

To continue the conversation around style, we only want to include changes that are necessary to resolve the issues. I've indicated which changes should be reverted.

**Review by staceyoue — APPROVED** [review]

**Review by rob-aitchison — APPROVED** [review]

**Comment by brettmaker** [comment]
Thanks for fixing this! is there a plan to push out a tag/release sometime soon?

**Comment by rob-aitchison** [comment]
> Thanks for fixing this! is there a plan to push out a tag/release sometime soon?

Hi @brettmaker , we just released it now - thanks for participating, we appreciate all the community efforts, cheers!

https://community.shotgridsoftware.com/t/a-new-version-of-tk-multi-breakdown2-has-been-released/15299/5
